File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    vector<string> word = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
    
    string complete_str = "";
    string compare_str = "";
    
    for(int i=0; i<s.length(); i++){
        if(s[i] >= '0' && s[i] <= '9'){
            complete_str += s[i];
            continue;
        }
        
        compare_str += s[i];
        
        for(int j=0; j<word.size(); j++){
            if(compare_str == word[j]){
                complete_str += to_string(j);
                compare_str = "";
                break;
            }
        }
    }
    
    answer = stoi(complete_str);
    
    return answer;
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var answer = 0;
    
    let word = [ "zero",  "one",  "two",  "three",  "four",  "five",  "six",  "seven", "eight", "nine" ];
    
    let complete_str = "";
    let compare_str = "";
    for(let i=0; i<s.length; i++){
        if(s[i] >= '0' && s[i] <= '9'){
            complete_str += s[i];
            continue;
        }
        
        compare_str += s[i];
        
        for(let j=0; j<word.length; j++){
            if(compare_str === word[j]){
                complete_str += j.toString();
                compare_str = "";
                break;
            }
        }
    }
    
    answer = parseInt(complete_str, 10);
    
    return answer;
}